core/src/main/scala/org/apache/spark/input/WholeTextFileInputFormat.scala

```diff
@@ -53,6 +53,19 @@ private[spark] class WholeTextFileInputFormat
     val totalLen = files.map(file => if (file.isDirectory) 0L else file.getLen).sum
     val maxSplitSize = Math.ceil(totalLen * 1.0 /
       (if (minPartitions == 0) 1 else minPartitions)).toLong
+
+    // For small files we need to ensure the min split size per node & rack <= maxSplitSize
+    val config = context.getConfiguration
+    val minSplitSizePerNode = config.getLong(CombineFileInputFormat.SPLIT_MINSIZE_PERNODE, 0L)
+    val minSplitSizePerRack = config.getLong(CombineFileInputFormat.SPLIT_MINSIZE_PERRACK, 0L)
+
+    if (maxSplitSize < minSplitSizePerNode) {
+      super.setMinSplitSizeNode(maxSplitSize)
```

Comment:
Is there a point in even checking the configuration? Why not just set these to `0L` unconditionally?

Reply:
AFAIU If we set these to `0L` unconditionally, every time there is left over data which wasn't combined into a split, would result in its own split because minSplitSizePerNode is `0L`. 
This shouldn't be an issue for small no. of files. But if we have a large no. of small files which result in a similar situation, we will end up having more splits rather than combining these together to form lesser no. of splits.

Reply:
Also if a user specifies them via configs we are ensuring that these don't break the code. If we set them to `0L` where a user specifies them, we would end up breaking the code anyways as the way `CombineFileInputFormat` works is it checks to see if the setting is `0L` or not. If it is 0 it ends up picking the value from the config.  https://github.com/apache/hadoop/blob/release-2.8.2-RC0/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/input/CombineFileInputFormat.java#L182 So we would have to atleast set the config to avoid hitting the error.